hermes/src/api/ws.rs

```diff
@@ -350,22 +350,36 @@ impl Subscriber {
             .keys()
             .cloned()
             .collect::<Vec<_>>();
-        for update in crate::aggregate::get_price_feeds_with_update_data(
+
+        let updates = match crate::aggregate::get_price_feeds_with_update_data(
             &*self.store,
             &price_feed_ids,
             RequestTime::AtSlot(event.slot()),
         )
         .await
-        .map_err(|e| {
-            tracing::warn!(
-                "Failed to get price feeds {:?} with update data: {:?}",
-                price_feed_ids,
-                e
-            );
-            e
-        })?
-        .price_feeds
         {
+            Ok(updates) => updates,
+            Err(_) => {
+                // The error can only happen when a price feed was available
+                // and is no longer there as we check the price feed ids upon
+                // subscription. In this case we send an error message and
+                // close the connection.
+                self.sender
+                    .send(
+                        serde_json::to_string(&ServerResponseMessage::Err {
```

Comment:
Can't we just ignore the removed feeds? In the current solution, when a price feed is removed, all websocket connections need to be opened up again and we will miss a few slots in between.

Reply:
Yeah it's a fair point and I have changed the code to follow what you says. However I think it is an unintuitive business logic and might surprise consumers. I wouldn't do it if our data ingestion wasn't relying on this.

The better approach would be what you had suggested before, a subscribeToAll method (or a `*`) and it is intuitive for a subscription to all to keep the connection while a feed is removed. However I didn't do it because it is a big change and we are going to move away from this API soon.